code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -803,6 +815,12 @@
 	glass_name = "cherry shake"
 	glass_desc = "A cherry flavored milkshake."
 
+/datum/reagent/consumable/cherryshake/on_mob_life(mob/living/carbon/C)
+	if(isjellyperson(C))
+		if(C.blood_volume < BLOOD_VOLUME_NORMAL(C))
+			C.blood_volume = min(BLOOD_VOLUME_NORMAL(C), C.blood_volume + 4.0)
```

Comment:
Why ``+ 4.0`` here instead of scaling based on reagent amount?

Reply:
I looked at the code for Bloody Mary and it says that its for quickly restoring blood level's I believe?

Reply:
Okay, I'm going to leave it as it is. Since it procs the same as bloody mary which is for fast blood healing I believe.